src/lifo.zig

```diff
@@ -0,0 +1,216 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const constants = @import("./constants.zig");
+
+/// An intrusive last in/first out (LIFO, stack) linked list.
+/// The element type T must have a field called "next" of type ?*T.
+pub fn LIFOType(comptime T: type) type {
+    return struct {
+        const LIFO = @This();
+
+        head: ?*T = null,
+
+        count: u64 = 0,
+        count_max: u64,
+
+        // This should only be null if you're sure we'll never want to monitor `count`.
+        name: ?[]const u8,
+
+        // If the number of elements is large, the constants.verify check in push() can be too
+        // expensive. Allow the user to gate it. Could also be a comptime param?
+        verify_push: bool,
+
+        pub fn init(count_max: u64, name: ?[]const u8, verify_push: bool) LIFO {
```

Comment:
```suggestion
        pub fn init(options: struct {
            capacity: u64, 
            verify_push: bool,
        }) LIFO {
```

Let's use named args pattern here: bools, numeric arguments, and optionals are usually better when named, and here we have all three.

And, as our stack _is_ statically sized, I think we don't need the `name`. So perhaps for LIFO the right design would be to say:

```
capacity: union(enum) {
    bounded: u32,
    unbounded: struct { name: []const u8 }
}
```

Reply:
I opted for the named parameters and made `capacity` a mandatory parameter as we do not yet have an unbounded use case for `LIFO`.